core/clients/key_flow_test.go

```diff
@@ -113,6 +118,67 @@ func TestKeyFlowInit(t *testing.T) {
 	}
 }
 
+type MyCustomClaims struct {
+	Foo string `json:"foo"`
+}
+
+func TestSetToken(t *testing.T) {
+	tests := []struct {
+		name         string
+		tokenInvalid bool
+		refreshToken string
+		wantErr      bool
+	}{
+		{
+			name:         "ok",
+			tokenInvalid: false,
+			refreshToken: "refresh_token",
+			wantErr:      false,
+		},
+		{
+			name:         "invalid_token",
+			tokenInvalid: true,
+			refreshToken: "",
+			wantErr:      true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var accessToken string
+			var err error
+			if tt.tokenInvalid {
+				accessToken = "foo"
+			} else {
+				accessTokenJWT := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.RegisteredClaims{
+					ExpiresAt: jwt.NewNumericDate(time.Now().Add(24 * time.Hour))})
+				accessToken, err = accessTokenJWT.SignedString(testSigningKey)
+				if err != nil {
+					t.Fatalf("get test access token as string: %s", err)
+				}
+			}
+
+			c := &KeyFlow{}
+			err = c.SetToken(accessToken, tt.refreshToken)
+
+			if (err != nil) != tt.wantErr {
+				t.Errorf("KeyFlow.SetToken() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			if err == nil {
+				expectedKeyFlowToken := &TokenResponseBody{
+					AccessToken:  accessToken,
+					ExpiresIn:    int(time.Now().Add(24 * time.Hour).Unix()),
```

Comment:
Hm, not sure how this timestamp is matching the one defined above since `time.Now` is probably different between the two calls. Probably we should define the timestamp above in a variable and use it in both places

Reply:
The test was passing so I guess its fast enough that they match, but I've changed it to be more robust